Controllers/CDebugController.cpp

```diff
@@ -26,14 +28,110 @@ CDebugController::~CDebugController()
 
 void CDebugController::run()
 {
-    m_breather_light.toggle();
+    static float breathing_light_intensity = 0;
+    mp_hw->setBreathingLight(breathing_light_intensity);
+    breathing_light_intensity = breathing_light_intensity ? 0 : 100;
 }
 
 bool CDebugController::newCommand(ICommand *p_command,
                                   IComChannel *p_comchannel)
 {
-    return CController::newCommand(p_command,
-                                   p_comchannel);  // todo: temporary plug
+    bool b_command_recognised = false;
+    ICommand::command_error_code_t error_code;
+    /**
+     * Set mains power output.
+     *
+     * mains(channel, power);
+     */
+    if (p_command->getName()->compare("mains") == 0)
+    {
+        error_code = setMainsPower(p_command);
+        b_command_recognised = true;
+    }
+    else if (p_command->getName()->compare("?mains") == 0)
+    {
+        float mains_power;
+        error_code = getMainsPower(p_command, &mains_power);
+        if (error_code == ICommand::COMMAND_OK)
+        {
+            CController::s_scratch_pad = "Mains channel ";
+            etl::to_string((*p_command)[0],
+                           CController::s_scratch_pad,
+                           etl::format_spec(),
+                           true);
+            CController::s_scratch_pad += ": ";
+            etl::to_string(mains_power,
+                           CController::s_scratch_pad,
+                           etl::format_spec().precision(2),
+                           true);
+            p_comchannel->send(CController::s_scratch_pad);
+        }
+        b_command_recognised = true;
+    }
+    if (b_command_recognised)
+    {
+        sendResultMessage(error_code, p_comchannel);
+    }
+    return (b_command_recognised);
 }
 
 void CDebugController::reset() {}
+
+ICommand::command_error_code_t CDebugController::setMainsPower(
+    ICommand *p_command)
+{
+    uint8_t mains_channel;
+    float mains_power;
+    // extract arguments from command.
+    if (p_command->getArgumentCount() == 1)
+    {
+        mains_channel = 0;
+        mains_power = (*p_command)[0];
+    }
+    else if (p_command->getArgumentCount() == 2)
+    {
+        mains_channel = (uint8_t)(*p_command)[0];
+        if (mains_channel != (*p_command)[0])
+        {
+            return (ICommand::ERROR_TYPE_MISMATCH);
+        }
+        mains_power = (*p_command)[1];
+    }
+    // range check arguments.
+    if ((mains_channel > 2) || (mains_power < 0) || (mains_power > 100))
```

Comment:
I think we need to fix this because if `_command->getArgumentCount()` is not 1 or 2, then mains_channel  and mains_power will be uninitialised.

Reply:
That's because I forgot final `else{}`. Fixed.